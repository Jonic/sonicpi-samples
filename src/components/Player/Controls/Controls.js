import React, { Component } from 'react'
import PropTypes from 'prop-types'

import PlayStop from './PlayStop/PlayStop'
import LoopCheckbox from './LoopCheckbox/LoopCheckbox'
import SampleSelector from './SampleSelector/SampleSelector'

class Controls extends Component {
  render() {
    return (
      <div className="c-Controls">
        <PlayStop
          isPlaying={this.props.isPlaying}
          playStopClickHandler= {this.props.playStopClickHandler}
        />

        <LoopCheckbox
          isLooping={this.props.isLooping}
          loopCheckboxClickHandler={this.props.loopCheckboxClickHandler}
        />

        <SampleSelector
          currentSample={this.props.currentSample}
          sampleSelectorChangeHandler={this.props.sampleSelectorChangeHandler}
          samples={this.props.samples}
        />
      </div>
    )
  }
}

Controls.propTypes = {
  currentSample:               PropTypes.string.isRequired,
  isPlaying:                   PropTypes.bool.isRequired,
  isLooping:                   PropTypes.bool.isRequired,
  loopCheckboxClickHandler:    PropTypes.func.isRequired,
  playStopClickHandler:        PropTypes.func.isRequired,
  sampleSelectorChangeHandler: PropTypes.func.isRequired,
  samples:                     PropTypes.array.isRequired,
}

export default Controls
