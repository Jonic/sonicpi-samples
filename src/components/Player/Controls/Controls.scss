@import '../../../styles/component-helper';

.c-controls {
}
