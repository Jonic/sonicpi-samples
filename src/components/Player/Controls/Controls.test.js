import React from 'react'
import { shallow } from 'enzyme'
import Controls from './Controls'

const categories = []
const categorySelectorChangeHandler = jest.fn()
const currentCategory = ''
const currentSample = ''
const filteredSamples = []
const isLoading = false
const isLooping = false
const isPlaying = false
const loopCheckboxClickHandler = jest.fn()
const playStopClickHandler = jest.fn()
const sampleSelectorChangeHandler = jest.fn()
const updateIsLoading = jest.fn()

let wrapper

describe('<Controls />', () => {
  beforeEach(() => {
    wrapper = shallow(
      <Controls
        categories={categories}
        categorySelectorChangeHandler={categorySelectorChangeHandler}
        currentCategory={currentCategory}
        currentSample={currentSample}
        filteredSamples={filteredSamples}
        isLoading={isLoading}
        isLooping={isLooping}
        isPlaying={isPlaying}
        loopCheckboxClickHandler={loopCheckboxClickHandler}
        playStopClickHandler={playStopClickHandler}
        sampleSelectorChangeHandler={sampleSelectorChangeHandler}
        updateIsLoading={updateIsLoading}
      />
    )
  })

  it('renders without crashing', () => {
    expect(wrapper.find('.c-Controls')).toBePresent()
  })
})
