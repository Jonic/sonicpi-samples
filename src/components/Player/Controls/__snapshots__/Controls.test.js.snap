// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Controls /> Renders... matches its snapshot 1`] = `
<div
  className="c-controls"
>
  <CategorySelector
    categories={Array []}
    categorySelectorChangeHandler={[MockFunction]}
    currentCategory=""
    isLoading={false}
    updateIsLoading={[MockFunction]}
  />
  <SampleSelector
    currentSample=""
    filteredSamples={Array []}
    isLoading={false}
    sampleSelectorChangeHandler={[MockFunction]}
    updateIsLoading={[MockFunction]}
  />
  <LoopCheckbox
    isLoading={false}
    isLooping={false}
    loopCheckboxClickHandler={[MockFunction]}
  />
  <PlayStop
    isLoading={false}
    isPlaying={false}
    playStopClickHandler={[MockFunction]}
  />
</div>
`;
